### PR TITLE
Move WebHook to Watcher and connect RabbitMQ [#256]

### PR DESCRIPTION
**Pull request by hekeemje**

### CODE REVIEW

File: backend/EasyMeets.Core/EasyMeets.Core.WebAPI/Extentions/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -149,8 +144,8 @@ public static void AddFirebaseAdmin(this IServiceCollection services, IConfigura
                 serviceAccount.PrivateKeyId = configuration["Firebase_Service_Account_Private_Id"];
                 serviceAccount.PrivateKey = configuration["Firebase_Service_Account_Private_Key"].Replace(@"\n", "\n");
             }
-            
-            var jsonSerializerSettings = new JsonSerializerSettings
+
+                var jsonSerializerSettings = new JsonSerializerSettings
```
Comment (Ymkavi) [inline backend/EasyMeets.Core/EasyMeets.Core.WebAPI/Extentions/ServiceCollectionExtensions.cs]: Please remove this indent

File: backend/EasyMeets.Watcher/EasyMeets.Watcher.WebAPI/Extensions/ServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
+﻿using EasyMeets.RabbitMQ.Interface;
```
Comment (Ymkavi) [inline backend/EasyMeets.Watcher/EasyMeets.Watcher.WebAPI/Extensions/ServiceCollectionExtensions.cs]: Please remove this not required using

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/Queue/WebHookListenerService.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            using var scope = _provider.CreateScope();
+            var calendarService = scope.ServiceProvider.GetRequiredService<ICalendarsService>();
+
+            try
+            {
+                await calendarService.SyncChangesFromGoogleCalendar(messageString);
+            }
+            catch (Exception ex)
+            {
+                throw new Exception(ex.Message);
```
Comment (Ymkavi) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/Queue/WebHookListenerService.cs]: Catching exceptions this way seems useless. We have the GenericExceptionHandler middleware

**Review by Ymkavi — CHANGES_REQUESTED** [review]

**Review by Ymkavi — APPROVED** [review]

File: backend/EasyMeets.RabbitMQ/Service/ConsumerService.cs [hunk 4]
```diff
@@ -16,24 +16,45 @@ public ConsumerService(IConnection connection, ConsumerSettings settings)
...
+            if (_settings.SequentialFetch)
+            {
+                _channel.BasicQos(0, 1, false);
+            }
+
+            _channel.BasicConsume(_settings.QueueName, _settings.AutoAcknowledge, consumer);
         }
 
-        public void Listen(AsyncEventHandler<BasicDeliverEventArgs> messageReceivedHandler)
+        public void ListenAsync(AsyncEventHandler<BasicDeliverEventArgs> messageReceivedHandler)
```
Comment (tatianahutii) [inline backend/EasyMeets.RabbitMQ/Service/ConsumerService.cs:37]: Why do we need an async and not an async version?

File: backend/EasyMeets.Watcher/EasyMeets.Watcher.BLL/EasyMeets.Watcher.BLL.csproj [hunk 5]
```diff
@@ -7,7 +7,11 @@
     </PropertyGroup>
 
     <ItemGroup>
-        <PackageReference Include="MediatR" Version="10.0.1"/>
+        <PackageReference Include="MediatR" Version="10.0.1" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <ProjectReference Include="..\..\EasyMeets.RabbitMQ\EasyMeets.RabbitMQ.csproj" />
```
Comment (tatianahutii) [inline backend/EasyMeets.Watcher/EasyMeets.Watcher.BLL/EasyMeets.Watcher.BLL.csproj:14]: You need to add this to the dockerfile

File: backend/EasyMeets.Watcher/EasyMeets.Watcher.WebAPI/Controllers/WebHookController.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        public IActionResult SyncCalendars(string email, string verifyCode)
+        {
+            if (verifyCode != Environment.GetEnvironmentVariable("WebHookGoogleAuthorizationCode"))
+            {
+                return Unauthorized();
+            }
+
+            _webHookNotifier.NotifyCalendarChanges(email);
+
+            return Ok();
```
Comment (tatianahutii) [inline backend/EasyMeets.Watcher/EasyMeets.Watcher.WebAPI/Controllers/WebHookController.cs]: This isn't CQRS architecture

File: backend/EasyMeets.Watcher/EasyMeets.Watcher.WebAPI/EasyMeets.Watcher.WebAPI.csproj [hunk 7]
```diff
@@ -4,15 +4,19 @@
...
+        <PackageReference Include="MediatR.Extensions.Microsoft.DependencyInjection" Version="10.0.1" />
+        <PackageReference Include="RabbitMQ.Client" Version="6.4.0" />
+        <PackageReference Include="Swashbuckle.AspNetCore" Version="6.2.3" />
     </ItemGroup>
 
     <ItemGroup>
-        <ProjectReference Include="..\EasyMeets.Watcher.BLL\EasyMeets.Watcher.BLL.csproj"/>
+        <ProjectReference Include="..\..\EasyMeets.Emailer\EasyMeets.Emailer.WebAPI\EasyMeets.Emailer.WebAPI.csproj" />
+        <ProjectReference Include="..\..\EasyMeets.RabbitMQ\EasyMeets.RabbitMQ.csproj" />
+        <ProjectReference Include="..\EasyMeets.Watcher.BLL\EasyMeets.Watcher.BLL.csproj" />
```
Comment (tatianahutii) [inline backend/EasyMeets.Watcher/EasyMeets.Watcher.WebAPI/EasyMeets.Watcher.WebAPI.csproj]: You need to add these to the Dockerfile

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Review by tatianahutii — APPROVED** [review]

**Review by apofis96 — APPROVED** [review]

**Pull request merged**